bluepill/src/BPRunner.m

```diff
@@ -190,24 +190,31 @@ - (int)runWithBPXCTestFiles:(NSArray<BPXCTestFile *> *)xcTestFiles {
         [BPUtils printInfo:ERROR withString:@"Packing failed: %@", [error localizedDescription]];
         return 1;
     }
-    if (bundles.count < numSims) {
-        [BPUtils printInfo:WARNING
-                withString:@"Lowering number of parallel simulators from %lu to %lu because there aren't enough test bundles.",
-                            numSims, bundles.count];
-        numSims = bundles.count;
-    }
     if (self.config.cloneSimulator) {
         self.testHostSimTemplates = [bpSimulator createSimulatorAndInstallAppWithBundles:xcTestFiles];
         if ([self.testHostSimTemplates count] == 0) {
             return 1;
         }
     }
+
+    int repeatTestsCount = [self.config.repeatTestsCount intValue];
+    if (repeatTestsCount > 1) {
```

Comment:
With the goal of handling retris inside of bluepill, I think we can move this logic into the main runloop. e.g. When a test fails, simply add `repeatTestsCount` bundles. Then disable retries in the subprocess

Reply:
This change is not explicitly targeting failure retries. While there might be an option to solve both issues in the same way, this simply enables running a test multiple times, regardless of if the first run passes or fails.